test/html_reload_test.rb

```diff
@@ -4,9 +4,37 @@ class HtmlReloadTest < ApplicationSystemTestCase
   test "reload HTML changes" do
     visit root_path
     assert_no_text "This was replaced"
+    assert_config("html-reload-strategy", "morph")
 
     edit_file "app/views/home/show.html.erb", replace: "_REPLACE_", with: "This was replaced!"
 
     assert_text "This was replaced"
   end
+
+  test "reload HTML changes with Turbo" do
+    with_turbo_reload_strategy do
```

Comment:
Instead of a block configuring the strategy, I prefer this approach:

- In `test_helper.rb`, add a common `setup` step that always resetst the option to its default of morph.
- In the tests, in a regular setup: set explicitly the option you want to test (morph in one case, replace in the other).

Also, I'd separate this into two test files:

- replace_reload_test.rb
- morph_reload_test.rb

Finally, for the replace reload test, you can consider adding some `<script></script>` tag that listens for a turbo load event and adds an attribute `data-turbo-navigated` to the body. That way, the test can check that this attribute was added, and that the expected logic triggered. The "morph" variant can check the absence of that attribute. You can add the script tag to `home/show` or even to the layout template in the dummy test app.